Microsoft.O365.Security.Native.ETW/UserTrace.hpp

```diff
@@ -73,6 +73,36 @@ namespace Microsoft { namespace O365 { namespace Security { namespace ETW {
         /// </example>
         virtual void Enable(O365::Security::ETW::RawProvider ^provider);
 
+        /// <summary>
+        /// Sets the trace properties for a session.
+        /// Must be called before Start().
+        /// See https://docs.microsoft.com/en-us/windows/win32/etw/event-trace-properties
+        /// for important details and restrictions.
+        ///
+        /// Configurable properties are ->
+        ///  - BufferSize. In KB. The maximum buffer size is 1024 KB.
+        ///  - MinimumBuffers. Minimum number of buffers is two per processor* .
+        ///  - MaximumBuffers.
+        ///  - FlushTimer. How often, in seconds, the trace buffers are forcibly flushed.
+        ///  - LogFileMode. EVENT_TRACE_NO_PER_PROCESSOR_BUFFERING simulates a *single* sequential processor.
+        /// </summary>
+        /// <param name="properties">the <see cref="O365::Security::ETW::EventTraceProperties"/> to set on the trace</param>
+        /// <example>
+        ///     var trace = new UserTrace();
+        ///     var properties = new EventTraceProperties
+        ///     {
+        ///         BufferSize = 256,
+        ///         LogFileMode = (uint)LogFileModeFlags.FLAG_EVENT_TRACE_REAL_TIME_MODE
+        ///     };
+        ///     trace.SetTraceProperties(properties);
+        ///     // ...
+        ///     trace.Open();
+        /// </example>
+        virtual void SetTraceProperties(EventTraceProperties ^properties);
+
+        // currently used in test scenarios only
+        virtual void Open();
```

Comment:
let's add a real comment on `Open()` that describes what it does -- bonus points if you include why an engineer may want to call this instead of (or before) Start()

Reply:
This saved me a few lines of P/Invoke in the test case. :-)

https://github.com/microsoft/krabsetw/pull/54 didn't add Open() for C++/CLI so I figured it was non-trivial to do so. Looking now, it's fairly straight-forward...expose Process() and add a wrapper class for EVENT_TRACE_LOGFILE - but I'm not sure that it adds much value? Which fields are worthwhile to expose?

Looking at the rationale in the original PR, that contributor wanted to programmatically check EVENT_TRACE_LOGFILE values (specifically ProcessTraceMode). But this value is fixed for krabs traces, and none of the other fields look relevant. So this seems to just be future proofing against library changes??

What am I missing?

Reply:
I think #54 didn't add Open() to C++/CLI because the contributor didn't have a need for it -- but that was a miss on our part during the code review (we should be maintaining consistency across C++ and C++/CLI interfaces).

It feels awkward for the C++ API to return `EVENT_TRACE_LOGFILE` but for the C++/CLI wrapper to return `void`. However, looking at `EVENT_TRACE_LOGFILE` I agree that it's unclear what would be valuable to promote to the C++/CLI wrapper class. 

I'm okay with deferring that work and continuing to return `void` here, but let's update the comment to match the C++ interface (`Opens a trace session.`)

Reply:
I also added Process() - since otherwise Open() is lonely.
And I added both to KernelTrace as well.